app/services/process_interaction_service.rb

```diff
@@ -123,7 +123,9 @@ def add_post!(body, account)
   end
 
   def status(xml)
-    Status.find(TagManager.instance.unique_tag_to_local_id(activity_id(xml), 'Status'))
+    uri = activity_id(xml)
+    return nil unless TagManager.instance.local_id?(uri)
+    Status.find(TagManager.instance.unique_tag_to_local_id(uri, 'Status'))
```

Comment:
This is kind of an edge case but unique_tag_to_local_id doesn't actually confirm that the URI belongs to this instance, only local_id? does that. Lots of important endpoints check local_id? properly, but not this one for favourites. So without checking this, a favourite could be created for a completely unrelated status just by incidentally having the same numeric ID part (this happened and was very confusing for the involved parties)